opendevin/core/config.py

```diff
@@ -107,11 +107,13 @@ class AgentConfig(metaclass=Singleton):
         name: The name of the agent.
         memory_enabled: Whether long-term memory (embeddings) is enabled.
         memory_max_threads: The maximum number of threads indexing at the same time for embeddings.
+        is_autonomous: Whether the agent is autonomous.
     """
 
     name: str = 'CodeActAgent'
     memory_enabled: bool = False
     memory_max_threads: int = 2
+    is_autonomous: bool = False
```

Comment:
And why do we need this? I think `is_autonoumous` can be entirely handled by the FE - no?
e.g., when the autonomous mode is enabled, whenever the agent is requesting user inputs, the FE will always send back a pre-defined user message, asking the agent to "continue". We can probably re-use the continue button logic - the "checkbox" will just work by keep clicking the "continue" button for the user.

Reply:
Removed the config and added it as an attribute to Agent.  The Agent should behave differently when autonomous mode is enabled like when to detect the loop.

Reply:
Why we need to change the backend for the autonomous mode?

ref: https://github.com/OpenDevin/OpenDevin/pull/2435#discussion_r1664446111

Reply:
for different prompts?

Reply:
what do you mean by "for different prompts"? can you elaborate?

Reply:
for different system prompts.

Reply:
why? I don't want us to add unnecessary additional complexity to the repo unless there are a few specific examples showing the necessity. The "continue" button on FE does not need to be modified on the backend; why does this need to modify BE?